src/lib/y2network/sysconfig/interface_file.rb

```diff
@@ -244,6 +244,16 @@ def variable_name(param_name)
       #   @return [String] IPOIB mode ("connected" or "datagram")
       define_variable(:ipoib_mode)
 
+      ## VLAN
+
+      # !@attribute [r] etherdevice
+      #   @return [String] Real device for the virtual LAN
+      define_variable(:etherdevice)
```

Comment:
Hmm, I am not sure here if String is good. Still how renaming should work?

Consider here. We have physical device A that is renamed to B. And we have old vlan, which has in etherdevice A and new vlan, which has there B. And now consider that we have device that is B and is renamed to A.

How this will work in writer? Always use original name of interface?

Reply:
ah, this should not go here, but to Vlan ConnectionConfig

Reply:
Exactly, I was confused by the comment xDD This is the interface file.

Reply:
> Hmm, I am not sure here if String is good. Still how renaming should work?
> 
> Consider here. We have physical device A that is renamed to B. And we have old vlan, which has in etherdevice A and new vlan, which has there B. And now consider that we have device that is B and is renamed to A.

That is what we were discussing about, if we do not reference with a Interface directly, then we will need a `service object` or `callback` or whatever that takes the responsibility of updating interface name references when an interface is renamed. 

Probably we should modify the parent device by a `ParentVlan` object that could be an interface, a connection uuid or a mac like NM permits. But for the time being I just used a simple String there.
> 
> How this will work in writer? Always use original name of interface?

Reply:
I actually prefer to have there Interface. Only part that I do not like previously was that class is responsible for that name to interface translation. It should be done by Config reader, that reads interfaces, connection configs, create necessary interfaces if needed and result should be connection config with proper interfaces. So it should be reader responsibility to do linking to interface as during read we are sure names are correct, later it can change.

Reply:
The problem is that there is a chance that, when the reader for a connection configuration runs, the underlying interface does not exist yet (think of virtual interfaces or even related connection config objects).

In such a case, we need to do a second pass. I would say that for that second pass, there are two options:

* Extending ConnectionConfig classes to know how to link its interfaces (a `#link` method could get the list of known interfaces and sets its own links).
* Extending readers with this logic.

Is this OK?

Reply:
@imobachgs yes, for me the preferred solution is extending reader with logic as linking is from my POV responsibility of reader and not connection config, which will have other business.